product/manager.go

```diff
@@ -245,17 +245,19 @@ func (m *manager) pollingClosure(apiURL url.URL) func(ctx context.Context) error
 				log.Debugf("product polling canceled, exiting")
 				return nil
 			}
-			p := res.APIProducts[i]
+			p := &res.APIProducts[i]
 			// Only match against proxy name when dealing with Proxy types
 			// We can't rely on p.OperationGroup.OperationConfigType because it
-			// may not be set if there are no operations so we have to do this
+			// will not be set if there are no operations so we have to do this
 			// here where we can check m.operationConfigTypes instead of in the
 			// Unmarshal function.
-			if (p.OperationGroup == nil || len(p.OperationGroup.OperationConfigs) == 0) &&
+			if p.OperationGroup == nil &&
 				strings.Contains(m.operationConfigTypes, ProxyOperationConfigType) {
-				p.APIs = append(p.APIs, p.Proxies...)
+				for _, proxy := range p.Proxies {
```

Comment:
In the case when we put `proxies` into `APIs`, should we remove the existing entires which are unmarshaled from `apigee-remote-service-targets` attributes? I think these two cases are exclusive but the products callout certainly will not filter out the attributes if it's configured to return products with op group typed "proxy".

Reply:
Sure. I can't see much benefit to keeping this as a feature.